packages/headless/src/features/insight-search/insight-analytics-actions.ts

```diff
@@ -0,0 +1,74 @@
+import {InsightPanel} from '@coveo/relay-event-types';
+import {
+  requiredNonEmptyString,
+  validatePayload,
+} from '../../utils/validate-payload';
+import {
+  InsightAction,
+  makeInsightAnalyticsAction,
+} from '../analytics/analytics-utils';
+import {getCaseContextAnalyticsMetadata} from '../case-context/case-context-state';
+
+export interface CreateArticleMetadata {
+  articleType: string;
+}
+
+export const logExpandToFullUI = (
+  caseId: string,
+  caseNumber: string,
+  fullSearchComponentName: string,
+  triggeredBy: string
+): InsightAction =>
+  makeInsightAnalyticsAction({
+    prefix: 'analytics/expandToFullUI',
+    __legacy__getBuilder: (client, state) => {
+      const meta = {
+        caseId,
+        caseNumber,
+        fullSearchComponentName,
+        triggeredBy,
+        caseContext: state.insightCaseContext?.caseContext || {},
+      };
+      return client.logExpandToFullUI(meta);
+    },
+    analyticsType: 'InsightPanel.ExpandToFullUI',
+    analyticsPayloadBuilder: (_state): InsightPanel.ExpandToFullUI => {
+      return {
+        context: {
+          targetId: caseId,
+          targetType: 'Case',
+          caseNumber: caseNumber,
```

Comment:
Why are we getting this from the parameter and not from the state? Looks like this is also available in the state caseContext no?